File: app/src/main/java/ru/goodibunakov/prodvtest/utils/HawkHelper.java
package ru.goodibunakov.prodvtest.utils;

import android.content.Context;

import com.orhanobut.hawk.Hawk;

import java.util.List;

import ru.goodibunakov.prodvtest.model.CityModel;

public class HawkHelper {

    public static final String ITEMS = "items";

    private static HawkHelper instance;

    public void init(Context context) {
        Hawk.init(context).build();
    }

    public List<CityModel> getItem(String key) {
        if (Hawk.contains(key))
            return Hawk.get(key);
        return null;
    }

    public boolean setItem(String key, List<CityModel> obj) {
        return Hawk.put(key, obj);
    }

    public boolean checkIfContain(String key) {
        return Hawk.contains(key);
    }

    public static HawkHelper getInstance() {
        if (instance != null)
            return instance;
        else return new HawkHelper();
    }
}

File: app/src/main/java/ru/goodibunakov/prodvtest/utils/HawkHelper.kt
package ru.goodibunakov.prodvtest.utils

import android.content.Context
import com.orhanobut.hawk.Hawk
import ru.goodibunakov.prodvtest.presentation.model.CityModel

object HawkHelper {

    const val ITEMS = "items"

    fun init(context: Context) {
        Hawk.init(context).build()
    }

    fun getItem(key: String): List<CityModel>? {
        return if (Hawk.contains(key)) Hawk.get<List<CityModel>>(key) else null
    }

    fun setItem(key: String, obj: List<CityModel>): Boolean {
        return Hawk.put(key, obj)
    }

    fun checkIfContain(key: String): Boolean {
        return Hawk.contains(key)
    }
}